### PR TITLE
Fix mobile worker creation

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?234275#1201037

The knockout validation wasn't showing an error before, so I moved it to angular. A bit of code duplication, but hopefully will be a little clearer when reading mobile worker code in teh future

@calellowitz 

@snopoke

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/registration/static/registration/js/password.js, line 8 - 'zxcvbn' is not defined. (no-undef)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 193 - 'password' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 197 - 'password' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 316 - 'djangoRMI' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 325 - 'zxcvbn' is not defined. (no-undef)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 337 - Missing trailing comma. (comma-dangle)
* corehq/apps/users/forms.py, line 387 - E501 line too long (156 > 115 characters)
* corehq/apps/users/forms.py, line 554 - E501 line too long (156 > 115 characters)

File: corehq/apps/users/templates/users/edit_commcare_user.html [hunk 1]
```diff
@@ -46,6 +46,8 @@
 
                         alert_user("{% trans 'Password changed successfully' %}", 'success');
                         ga_track_event("Edit Mobile Worker", "Reset password", "{{couch_user_id|escapejs}}");
+                    } else {
+                        alert_user("{% trans 'Password was not changed' %}", 'danger');
```
Comment (emord) [inline corehq/apps/users/templates/users/edit_commcare_user.html]: Not a great error. we don't currently show an error even when the passwords don't match with each other. would take a bit more refactoring that gives an exact error to the user.

File: corehq/apps/users/forms.py [hunk 2]
```diff
@@ -545,10 +545,13 @@ def __init__(self, project, *args, **kwargs):
...
-            self.fields['password'].widget = forms.TextInput()
+            self.fields['password'].widget = forms.TextInput(attrs={
+                "validate_password": "",
+                "ng_keydown": "markNonDefault()",
+                "class": "default",
+            })
             self.fields['password'].help_text = mark_safe("""
-                <i class="fa fa-warning"></i> This password will not be shown again. <br />
-                <span data-bind="text: passwordHelp, css: color">
+                <i class="fa fa-warning"></i>This password is automatically generated. Please copy it or create your own. It will not be shown again. <br />
```
Comment (snopoke) [inline corehq/apps/users/forms.py]: Translation

File: corehq/apps/users/static/users/js/mobile_workers.ng.js [hunk 3]
```diff
@@ -33,7 +39,47 @@
...
                 .addClass('has-error');
-        }
+        },
+        passwordSuccess: function () {
+            $formElements.password()
+                .removeClass('has-error has-pending')
+                .addClass('has-success');
+            if ($formElements.password().hasClass('non-default')) {
+                $formElements.passwordHint()
+                    .text('Good Job! Your password is strong!');
```
Comment (snopoke) [inline corehq/apps/users/static/users/js/mobile_workers.ng.js]: Translation, and below

File: corehq/apps/users/static/users/js/mobile_workers.ng.js [hunk 4]
```diff
@@ -259,6 +313,31 @@
...
+                ctrl.$validators.validatePassword = function (password) {
+                    if (!password) {
+                        return false;
+                    }
+                    var score = zxcvbn(password, ['dimagi', 'commcare', 'hq', 'commcarehq']).score,
+                        goodEnough = score > 1;
+
+                    if (score < 1) {
+                        visualFormCtrl.passwordError();
+                    } else if (score > 1){
```
Comment (snopoke) [inline corehq/apps/users/static/users/js/mobile_workers.ng.js]: Could reuse the `goodEnough` variable here

**Comment by dimagimon** [comment]
There are 9 errors:

* corehq/apps/registration/static/registration/js/password.js, line 8 - 'zxcvbn' is not defined. (no-undef)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 195 - 'password' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 199 - 'password' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 318 - 'djangoRMI' is defined but never used (no-unused-vars)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 327 - 'zxcvbn' is not defined. (no-undef)
* corehq/apps/users/static/users/js/mobile_workers.ng.js, line 340 - Missing trailing comma. (comma-dangle)
* corehq/apps/users/forms.py, line 387 - E501 line too long (139 > 115 characters)
* corehq/apps/users/forms.py, line 388 - E126 continuation line over-indented for hanging indent
* corehq/apps/users/forms.py, line 554 - E501 line too long (138 > 115 characters)

**Comment by emord** [comment]
:white_check_mark: (ignoring dimagimon as those are issues that existed before)

**Pull request merged**